floris/tools/uncertainty_interface.py

```diff
@@ -1,337 +1,174 @@
-
-        if self.fix_yaw_in_relative_frame:
-            # The relative yaw angle is fixed and always has the nominal
-            # value (e.g., 0 deg) when evaluating uncertainty. Evaluating
-            # wind direction uncertainty like this would essentially come
-            # down to a Gaussian smoothing of FLORIS solutions over the
-            # wind directions. This can also be really fast, since it would
-            # not require any additional calculations compared to the
-            # non-uncertainty FLORIS evaluation.
-            yaw_angles_probablistic = np.vstack([
-                np.expand_dims(yaw_angles_nominal, axis=0)
-                for _ in unc_pmfs["wd_unc"]
-            ])
-        else:
-            # Fix yaw angles in the absolute (compass) reference frame,
-            # meaning that for each probablistic wind direction evaluation,
-            # our probablistic (relative) yaw angle evaluated goes into
-            # the opposite direction. For example, a probablistic wind
-            # direction 3 deg above the nominal value means that we evaluate
-            # it with a relative yaw angle that is 3 deg below its nominal
-            # value.
-            yaw_angles_probablistic = np.vstack([
-                np.expand_dims(yaw_angles_nominal, axis=0) - dy
-                for dy in unc_pmfs["wd_unc"]
-            ])
-
-        self.wd_array_probablistic = wd_array_probablistic
-        self.yaw_angles_probablistic = yaw_angles_probablistic
-
-    def _reassign_yaw_angles(self, yaw_angles=None):
-        # Overwrite the yaw angles in the FlorisInterface object
-        if yaw_angles is not None:
-            self.fi.floris.farm.yaw_angles = yaw_angles
-
-    # Public methods
+        # Get the weights
+        self.weights = self._get_weights(self.wd_std, self.wd_sample_points)
 
-    def copy(self):
-        """Create an independent copy of the current UncertaintyInterface
-        object"""
-        fi_unc_copy = copy.deepcopy(self)
-        fi_unc_copy.fi = self.fi.copy()
-        return fi_unc_copy
+        # Instantiate the FlorisInterface
+        self.floris_interface = FlorisInterface(configuration)
 
-    def reinitialize_uncertainty(
+    def copy(self):
+        """Create an independent copy of the current FlorisInterface object"""
+        return UncertaintyInterface(self.floris_interface.floris.as_dict(),
+                                    wd_resolution=self.wd_resolution,
+                                    ws_resolution=self.ws_resolution,
+                                    ti_resolution=self.ti_resolution,
+                                    yaw_resolution=self.yaw_resolution,
+                                    power_setpoint_resolution=self.power_setpoint_resolution,
+                                    wd_std=self.wd_std,
+                                    wd_sample_points=self.wd_sample_points,
+                                    verbose=self.verbose)
+
+
+    def set(
```

Comment:
I wonder if we can pass `**args` directly through to `FlorisInterface.set()` and point to `FlorisInterface.set()` in the docstring here. It's a nitpick, but it seems like a lot of repeated code that might get out of sync easily.